services/wiki/wiki.go

```diff
@@ -0,0 +1,331 @@
+// Copyright 2015 The Gogs Authors. All rights reserved.
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package wiki
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/sync"
+
+	"github.com/unknwon/com"
+)
+
+var (
+	reservedWikiNames = []string{"_pages", "_new", "_edit", "raw"}
```

Comment:
if we use a map["string"]bool lookup could be faster?

like zeripath mentioned in https://github.com/go-gitea/gitea/pull/9220#discussion_r355133946

Reply:
or at least use `util.IsStringInSlice` but i find the map version more elegant

Reply:
@6543 It will not because the items is too less.

Reply:
ok so then i would sugessd util.IsStringInSlice simply because of code redundancy

Reply:
```diff
index b7a2b95f4..59c681164 100644
--- a/services/wiki/wiki.go
+++ b/services/wiki/wiki.go
@@ -15,6 +15,7 @@ import (
        "code.gitea.io/gitea/modules/git"
        "code.gitea.io/gitea/modules/log"
        "code.gitea.io/gitea/modules/sync"
+       "code.gitea.io/gitea/modules/util"
 
        "github.com/unknwon/com"
 )
@@ -25,11 +26,9 @@ var (
 )
 
 func nameAllowed(name string) error {
-       for _, reservedName := range reservedWikiNames {
-               if name == reservedName {
-                       return models.ErrWikiReservedName{
-                               Title: name,
-                       }
+       if util.IsStringInSlice(name, reservedWikiNames) {
+               return models.ErrWikiReservedName{
+                       Title: name,
                }
        }
        return nil
```

Reply:
Optimize could be don later ...